### DISCUSSION TITLE
Transport Protocol Change in Fast DDS

### DISCUSSION THREAD

**Discussion by AnoopGQuest** [eProsima/Fast-DDS#4900] [General]
Can we send data from publisher side using a particular protocol like UDP and receive data from the subscriber side using another protocol like TCP?

I was running a ros2 publisher node with transport protocol selected as UDP and ros2 subscriber node with transport protocol selected as PCIe...I was not able to subscribe the message sent by the publisher. Why?

**Comment by JesusPoderoso** [comment]
Hi @AnoopGQuest, thanks for using Fast DDS.
I am afraid that is not possible using Fast DDS: the subscriber and publishers won't match. But [DDS Router](https://eprosima-dds-router.readthedocs.io/en/latest/index.html) can fulfill your requirements (using the Fast DDS transports: SHM, UDP and/or TCP).

Although this would be the scenario using just Fast DDS...

```mermaid
graph LR;
    UDP_writer -. / .-> TCP_reader
``` 
The eProsima's [DDS Router](https://eprosima-dds-router.readthedocs.io/en/latest/index.html) can perform the communication.

```mermaid
graph LR;
    UDP_writer --> DDS_Router
    DDS_Router --> TCP_reader
``` 

[DDS Router](https://eprosima-dds-router.readthedocs.io/en/latest/index.html) allows [configuring different transports](https://eprosima-dds-router.readthedocs.io/en/latest/rst/user_manual/configuration.html#custom-transport-descriptors) on different DDS Router's participant, so your requested scenario can be addressed.

```mermaid
graph LR;
    UDP_writer --> UDP_participant
    TCP_participant --> TCP_reader
   subgraph DDS_Router
    UDP_participant-->TCP_participant
    end
```
